src/foam/lib/parse/ErrorReportingPStream.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.lib.parse;
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class ErrorReportingPStream
+    extends ProxyPStream
+{
+  public static final List<Character> ASCII_CHARS = IntStream.rangeClosed(0, 255)
+      .mapToObj(i -> (char) i)
+      .collect(Collectors.toList());
+
+  protected ThreadLocal<StringBuilder> sb = new ThreadLocal<StringBuilder>() {
+    @Override
+    protected StringBuilder initialValue() {
+      return new StringBuilder();
+    }
+
+    @Override
+    public StringBuilder get() {
+      StringBuilder b = super.get();
+      b.setLength(0);
+      return b;
+    }
+  };
+
+  protected Parser errParser = null;
+  protected ParserContext errContext = null;
+  protected ErrorReportingNodePStream errStream = null;
+
+  protected int pos;
```

Comment:
Is this variable needed? Won't 'pos' always be zero?

Reply:
Yeah I think so. Removing now.